### PR TITLE
Export `AbortSignal`

### PR DESCRIPTION
**Pull request by fregante**
Proposal. 

Blocked by:

- https://github.com/sindresorhus/dom-loaded/issues/10

### CODE REVIEW

**Comment by sindresorhus** [comment]
CI has been fixed.

File: index.d.ts [hunk 1]
```diff
@@ -15,15 +15,28 @@ declare const domLoaded: Promise<void> & {
...
 	if (domLoaded.hasLoaded) {
-		console.log('The DOM has already finished loading.')
+		console.log('The DOM has already finished loading.');
 	}
 	```
 	*/
 	readonly hasLoaded: boolean;
+
+	/**
+	AbortSignal that is aborted when the DOM has already finished loading.
```
Comment (sindresorhus) [inline index.d.ts]: It's unclear from the description what happens if the DOM has not already finished loading.

File: index.d.ts [hunk 2]
```diff
@@ -15,15 +15,28 @@ declare const domLoaded: Promise<void> & {
...
+
+	@example
+	```
+	import domLoaded from 'dom-loaded';
+
+	showLoader({signal: domLoaded.signal});
+	```
+	*/
+	readonly hasLoaded: boolean;
```
Comment (sindresorhus) [inline index.d.ts]: Not updated

File: index.d.ts (hunk 1 above)
Comment (sindresorhus) [reply index.d.ts]:
```suggestion
	An `AbortSignal` that triggers when the DOM finishes loading or immediately if it has already loaded.
```

File: index.d.ts (hunk 2 above)
Comment (fregante) [reply index.d.ts:39]:
```suggestion
	readonly signal: boolean;
```

File: index.d.ts [hunk 3]
```diff
@@ -15,15 +15,28 @@ declare const domLoaded: Promise<void> & {
...
+	An `AbortSignal` that triggers when the DOM finishes loading or immediately if it has already loaded.
+
+	@example
+	```
+	import domLoaded from 'dom-loaded';
+
+	showLoader({signal: domLoaded.signal});
+	```
+	*/
+	readonly signal: boolean;
```
Comment (sindresorhus) [inline index.d.ts]: It's not a boolean.

Comment (fregante) [reply index.d.ts:39]:
```suggestion
	readonly signal: AbortSignal;
```

**Pull request merged**